presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentControlUtils.java

```diff
@@ -0,0 +1,220 @@
+    }
+
+    public static Response stopPresto(StopType stopType)
+    {
+        if (!isInstalled()) {
+            return Response.status(NOT_FOUND).build();
+        }
+        if (!isRunning()) {
+            LOGGER.debug("Presto is not running");
+            return Response.status(OK).build();
+        }
+        try {
+            stopHelper(stopType);
+        }
+        catch (PrestoManagerException e) {
+            LOGGER.error(e.getMessage(), e.getCause());
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        catch (Exception e) {
+            LOGGER.error("Unhandled exception", e);
+            return Response.status(INTERNAL_SERVER_ERROR).build();
+        }
+        return Response.status(OK).build();
+    }
+
+    private static void stopHelper(StopType stopType)
+            throws PrestoManagerException
+    {
+        switch (stopType) {
+            case TERMINATE:
+                int prestoTerminate = executeCommand("service", "presto", "stop");
+                if (prestoTerminate != 0) {
+                    throw new PrestoManagerException("Failed to stop presto", prestoTerminate);
+                }
+                break;
+            case KILL:
+                executeCommand("sudo", LAUNCHER_SCRIPT, "kill");
+                int prestoKill = executeCommand("service", "presto", "status");
+                if (prestoKill != 3) {
+                    throw new PrestoManagerException("Failed to stop presto", prestoKill);
+                }
+                break;
+            case GRACEFUL:
+                try {
+                    if (getFileProperty(PATH_TO_CONFIG_FILE, "coordinator").equalsIgnoreCase("true")) {
+                        throw new PrestoManagerException("Coordinator can't be gracefully stopped");
+                    }
+                    String prestoPort = getFileProperty(PATH_TO_CONFIG_FILE, "http-server.http.port");
+                    Response response = newRequestBuilder(new URI(format("localhost:%s/v1/info/state", prestoPort)))
+                            .withBody(entity((new Gson().toJson("SHUTTING_DOWN")), APPLICATION_JSON_TYPE)).buildPut().invoke();
+                    if (response.getStatus() != 200) {
+                        throw new PrestoManagerException("Failed to stop presto gracefully");
+                    }
+                }
+                catch (IOException | URISyntaxException e) {
+                    throw new PrestoManagerException("Error in stopping presto", e);
+                }
+        }
+    }
+
+    public static Response restartPresto()
+    {
+        if (!isInstalled()) {
+            return Response.status(NOT_FOUND).build();
+        }
+        try {
+            int prestoRestart = executeCommand("service", "presto", "restart");
+            if (prestoRestart != 0) {
+                LOGGER.error("Failed to restart presto; Process exited with return value: {}", prestoRestart);
```

Comment:
You should throw exceptions here as well, otherwise, response OK may have been sent back